lib/jenkins_pipeline_builder/generator.rb

```diff
@@ -48,29 +47,25 @@ def view
 
     def bootstrap(path, project_name = nil)
       logger.info "Bootstrapping pipeline from path #{path}"
-      job_collection.load_from_path(path)
-      job_collection.remote_dependencies.cleanup
-
-      if job_collection.projects.any?
-        errors = publish_project(project_name)
-      else
-        errors = publish_jobs(job_collection.standalone_jobs)
-      end
-      print_compile_errors errors
-      errors
+      load_job_collection path
+      publish(project_name || job_collection.projects.first[:name])
     end
 
     def pull_request(path, project_name)
       logger.info "Pull Request Generator Running from path #{path}"
-      job_collection.load_from_path(path)
-      job_collection.remote_dependencies.cleanup
-      logger.info "Project: #{job_collection.projects}"
-      errors = {}
-      job_collection.projects.each do |project|
-        next unless project[:name] == project_name || project_name.nil?
-        errors.merge! process_pull_request_project project
+      load_job_collection path
+      defaults = job_collection.defaults[:value]
+      pr_generator = PullRequestGenerator.new application_name: defaults[:application_name],
+                                              git_url: defaults[:github_site],
+                                              git_org: defaults[:git_org],
+                                              git_repo: defaults[:git_repo_name]
+      pr_generator.delete_closed_prs unless JenkinsPipelineBuilder.debug
```

Comment:
I'd rather have the delete_closed_prs method know that it shouldn't if in debug mode.